src/planning/plan.cpp

```diff
@@ -96,80 +97,94 @@ class NodeCompare
 
 using pqueue_t = std::priority_queue<Node*, std::vector<Node*>, NodeCompare>;
 using set_t = std::set<Node*, NodeEqualityCompare>;
+using collides_predicate_t = std::function<bool(double x, double y , double radius)>;
```

Comment:
Technically whether a pose is in collision might also depend on orientation (since the rover's not shaped like a circle), but obviously we are very far from the point where we need to worry about that.